
---
title: "Dual RNA-seq of Lupin infected with P. Cinnamomi"
author: "Mark Ziemann"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
theme: cosmo
---

Source: https://github.com/markziemann/phytophthora-lupin

Vew the reports: http://118.138.234.73/public/barry/

## Intro

Phytophthora infected Lupinus angustifolius root.

## Reference genome info

Here is the reference information

Lupinus angustifolius from Ensembl
Lupinus_angustifolius.LupAngTanjil_v1.0.49.gtf.gz
Lupinus_angustifolius.LupAngTanjil_v1.0.dna_sm.toplevel.fa.gz

Phytophthora cinnamomi genomic information obtained from a new assembly found on genbank
https://ftp.ncbi.nlm.nih.gov/genomes/genbank/protozoa/Phytophthora_cinnamomi/latest_assembly_versions/GCA_018691715.1_ASM1869171v1/

fasta files were combined before indexing
gtf files were combined before indexing
index created by STAR

## Splitting genes

Based on the below genome annotation data, we classified genes with the names containing "Tanjil" and
"ENSRNA" as belonging to Lupin and genes containing "IUM83" belonging to lupin

(base) ziemannm@bio-g-2:~/barry/ref3$ grep -c Tanjil *gtf
combined.gtf:497300
GCA_018691715.1_ASM1869171v1_genomic.gtf:0
Lupinus_angustifolius.LupAngTanjil_v1.0.49.gtf:497300
(base) ziemannm@bio-g-2:~/barry/ref3$ grep -c IUM83 *gtf
combined.gtf:191926
GCA_018691715.1_ASM1869171v1_genomic.gtf:191926
Lupinus_angustifolius.LupAngTanjil_v1.0.49.gtf:0
(base) ziemannm@bio-g-2:~/barry/ref3$ grep -c ENS *gtf
combined.gtf:6291
GCA_018691715.1_ASM1869171v1_genomic.gtf:3
Lupinus_angustifolius.LupAngTanjil_v1.0.49.gtf:6288


## Data processing

Read trimming with Skewer v0.2.2 followed by STAR v2.7.7a.
STAR gene level counts (reverse strand) were imported into R.
Lupin and phytophthora counts were separated.
DESeq2 was run to quantify differences between treatment groups.

## Read in data


```{r,library}

#conda activate R40
library("DESeq2")
library("mitch")
library("reshape2")
library("Biostrings")
library("gplots")


```

Load counts into R

```{r,load}

tmp <- read.table("3col.tsv.gz")
y<-as.matrix(acast(tmp, V2~V1, value.var="V3"))
colnames(y) <- sapply(strsplit(colnames(y),"_"),"[[",1)
head(y)

```

Load samplesheet and change the colnames.

```{r,samplesheet}

ss <- read.table("samplesheet.tsv")
ss

ss$V1 == colnames(y)
colnames(y) <- ss$V2

```



QC analysis of counts and split by species.


```{r,split,fig.width=8,fig.height=8}

# make sure all genes are accounted for
dim(y)
length(grep("ENS",rownames(y)))
length(grep("Tanjil",rownames(y)))
length(grep("IUM83",rownames(y)))
length(grep("ENS",rownames(y))) + length(grep("Tanjil",rownames(y))) + length(grep("IUM83",rownames(y)))

par(mar=c(5,10,4,2))

pc <- y[grep("IUM83",rownames(y)),]
dim(pc)

la <- y[grep("IUM83",rownames(y),invert=TRUE),]
dim(la)

barplot(colSums(y),horiz=TRUE,las=1,xlab="total number of assigned reads")

barplot(colSums(pc),horiz=TRUE,las=1,xlab="number of PC assigned reads")

barplot(colSums(la),horiz=TRUE,las=1,xlab="number of LA assigned reads")

barplot(log10(colSums(pc)),horiz=TRUE,las=1,xlab="log10 number of PC assigned reads")

barplot(log10(colSums(la)),horiz=TRUE,las=1,xlab="log10 number of LA assigned reads")

barplot( colSums(pc)/colSums(y), horiz=TRUE,las=1,xlab="proportion of PC/total reads" )

# revert to defaults
par(mar=c(5.1,4.1,4.1,2.1))


```

## Multidimensional scaling analysis

```{r,mds,fig.width=8,fig.height=8}

mds <- cmdscale(dist(t(y)))
plot(mds*1.05, xlab="Coordinate 1", ylab="Coordinate 2", type = "p",pch=19,col="lightblue",bty="n")
text(mds, labels=colnames(y)) 
mtext("overall MDS plot")

mds <- cmdscale(dist(t(pc)))
plot(mds*1.05, xlab="Coordinate 1", ylab="Coordinate 2", type = "p",pch=19,col="lightblue",bty="n")
text(mds, labels=colnames(pc))
mtext("Pc MDS plot")

mds <- cmdscale(dist(t(la)))
plot(mds*1.05, xlab="Coordinate 1", ylab="Coordinate 2", type = "p",pch=19,col="lightblue",bty="n")
text(mds, labels=colnames(la))
mtext("La MDS plot")

```

## Stable genes

The goal here is to identify genes whose expression is stable over a range of samples to use as qPCR controls.


```{r,stable_pc}

pcx <- pc[,grep("P",colnames(pc))]
pcx <- pcx[which(rowMeans(pcx)>10),]

pcy <- pcx/colSums(pcx) * 1000000


pcy_means <- rowMeans(pcy)
pcy_cv <- apply(pcy,1,sd) / rowMeans(pcy) *100
pcy_logmeans <- log10(rowMeans(pcy))

hist(pcy_cv,xlab="%CoV")
hist(pcy_logmeans,xlab="log10 CPM")

plot(pcy_logmeans,pcy_cv,xlab="baseline expression (log10 CPM)",ylab="coefficient of variation (%)")
grid()
abline(h=110, v=2.5,col="red")

pcydf <- data.frame(pcy_logmeans,pcy_cv)

pcydf_subset <- subset(pcydf,pcy_logmeans>2.5 & pcy_cv<110)

pcy[which(rownames(pcy)  %in%  rownames(pcydf_subset)),]

pcydf_subset

```


## Contrasts 


P cinnamomi. These will be called Pc1 to Pc6

1. PcHY vs LaP18. "DE early"

2. PcHY vs LaP30. "DE mid"

3. PcHY vs LaP48. "DE late"

4. LaPc18 vs LaPc30 "Change from early to mid"

5. LaPc30 vs LaPc48 "Change from mid to late"

6. LaPc18 vs LaPc48 "Difference between early and late"


```{r,splitPCcontrasts}

ss$grp <- sapply(strsplit(ss$V2,"_"),"[[",1)
head(pc)

sspc1 <- subset(ss,grp=="PcHY-0" | grp=="LaP-18")
sspc1$trt <- as.numeric(sspc1$grp=="LaP-18")
pc1 <- pc[,which(colnames(pc) %in% sspc1$V2)] 
head(pc1)
head(sspc1)

sspc2 <- subset(ss,grp=="PcHY-0" | grp=="LaP-30")
sspc2$trt <- as.numeric(sspc2$grp=="LaP-30")
pc2 <- pc[,which(colnames(pc) %in% sspc2$V2)]
head(pc2)
head(sspc2)

sspc3 <- subset(ss,grp=="PcHY-0" | grp=="LaP-48")
sspc3$trt <- as.numeric(sspc3$grp=="LaP-48")
pc3 <- pc[,which(colnames(pc) %in% sspc3$V2)]
head(pc3)
head(sspc3)

sspc4 <- subset(ss,grp=="LaP-18" | grp=="LaP-30")
sspc4$trt <- as.numeric(sspc4$grp=="LaP-30")
pc4 <- pc[,which(colnames(pc) %in% sspc4$V2)]
head(pc4)
head(sspc4)

sspc5 <- subset(ss,grp=="LaP-30" | grp=="LaP-48")
sspc5$trt <- as.numeric(sspc5$grp=="LaP-48")
pc5 <- pc[,which(colnames(pc) %in% sspc5$V2)]
head(pc5)
head(sspc5)

sspc6 <- subset(ss,grp=="LaP-18" | grp=="LaP-48")
sspc6$trt <- as.numeric(sspc6$grp=="LaP-48")
pc6 <- pc[,which(colnames(pc) %in% sspc6$V2)]
head(pc6)
head(sspc6)

```


Lupin. These will be called La1 to La5

1. LaH-0 vs LaP18 "DE early"

2. LaH-0 vs LaP30 "DE mid"

3. LaH-0 vs LaP48 "DE late"

4. LaPc18 vs LaPc30 "Change from early to mid"

5. LaPc30 vs LaPc48 "Change from mid to late"

6. LaPc18 vs LaPc48 "Difference between early and late"



```{r,splitLScontrasts}

head(la)

ssla1 <- subset(ss,grp=="LaH-0" | grp=="LaP-18")
ssla1$trt <- as.numeric(ssla1$grp=="LaP-18")
la1 <- la[,which(colnames(pc) %in% ssla1$V2)]
head(la1)
head(ssla1)

ssla2 <- subset(ss,grp=="LaH-0" | grp=="LaP-30")
ssla2$trt <- as.numeric(ssla2$grp=="LaP-30")
la2 <- la[,which(colnames(pc) %in% ssla2$V2)]
head(la2)
head(ssla2)

ssla3 <- subset(ss,grp=="LaH-0" | grp=="LaP-48")
ssla3$trt <- as.numeric(ssla3$grp=="LaP-48")
la3 <- la[,which(colnames(pc) %in% ssla3$V2)]
head(la3)
head(ssla3)

ssla4 <- subset(ss,grp=="LaP-18" | grp=="LaP-30")
ssla4$trt <- as.numeric(ssla4$grp=="LaP-30")
la4 <- la[,which(colnames(pc) %in% ssla4$V2)]
head(la4)
head(ssla4)

ssla5 <- subset(ss,grp=="LaH-30" | grp=="LaP-48")
ssla5$trt <- as.numeric(ssla5$grp=="LaP-48")
la5 <- la[,which(colnames(pc) %in% ssla4$V2)]
head(la5)
head(ssla5)

ssla6 <- subset(ss,grp=="LaH-18" | grp=="LaP-48")
ssla6$trt <- as.numeric(ssla6$grp=="LaP-48")
la6 <- la[,which(colnames(pc) %in% ssla6$V2)]
head(la6)
head(ssla6)

```

## DE function

Here is a function for differential expression.

```{r,de_function}

run_de <- function(ss,xx){
y <- round(xx)
y <- y[rowMeans(y)>10,]
# MDS
mds <- cmdscale(dist(t(y)))
XMAX=max(mds[,1])*1.1
XMIN=min(mds[,1])*1.1
plot( mds , xlab="Coordinate 1", ylab="Coordinate 2",
  type = "n" , xlim=c(XMIN,XMAX),main="MDS plot",bty="n")
text(mds, labels=colnames(y) )
# DE
dds <- DESeqDataSetFromMatrix(countData=y, colData = ss, design = ~ trt)
dds <- DESeq(dds)
de <- DESeq2::results(dds)
de <- de[order(de$pvalue),]
up <- rownames(subset(de, log2FoldChange>0 & padj<0.05 ))
dn <- rownames(subset(de, log2FoldChange<0 & padj<0.05 ))
str(up)
str(dn)
# MA plot
sig <-subset(de, padj < 0.05 )
GENESUP <- length(up)
GENESDN <- length(dn)
SUBHEADER = paste(GENESUP, "up, ", GENESDN, "down")
ns <-subset(de, padj > 0.05 )
plot(log2(de$baseMean),de$log2FoldChange,
     xlab="log2 basemean", ylab="log2 foldchange",
     pch=19, cex=0.5, col="dark gray",
     main="smear plot")
points(log2(sig$baseMean),sig$log2FoldChange,
       pch=19, cex=0.5, col="red")
mtext(SUBHEADER)
# heatmap
yn <- y/colSums(y)*1000000
yf <- yn[which(rownames(yn) %in% rownames(de)[1:50]),]
mycols <- gsub("0","yellow",ss$trt)
mycols <- gsub("1","orange",mycols)
colfunc <- colorRampPalette(c("blue", "white", "red"))
heatmap.2(  as.matrix(yf), col=colfunc(25),scale="row",
    ColSideColors =mycols ,trace="none",
    margin = c(10,10), cexRow=0.6, cexCol=0.8 , main="Top 50 genes by p-val")
mtext("yellow=ctrl, orange=trt")
return(de)
}

```

## Run differential expression

Here execute the analysis.

```{r,de_run,fig.width=8,fig.height=8}

pc1de <- run_de(sspc1,pc1)
head(as.data.frame(pc1de),20)
write.table(pc1de,file="pc1de.tsv",quote=FALSE,sep="\t")

pc2de <- run_de(sspc2,pc2)
head(as.data.frame(pc2de),20)
write.table(pc2de,file="pc2de.tsv",quote=FALSE,sep="\t")

pc3de <- run_de(sspc3,pc3)
head(as.data.frame(pc3de),20)
write.table(pc3de,file="pc3de.tsv",quote=FALSE,sep="\t")

pc4de <- run_de(sspc4,pc4)
head(as.data.frame(pc4de),20)
write.table(pc4de,file="pc4de.tsv",quote=FALSE,sep="\t")

pc5de <- run_de(sspc5,pc5)
head(as.data.frame(pc5de),20)
write.table(pc5de,file="pc5de.tsv",quote=FALSE,sep="\t")

la1de <- run_de(ssla1,la1)
head(as.data.frame(la1de),20)
write.table(la1de,file="la1de.tsv",quote=FALSE,sep="\t")

la2de <- run_de(ssla2,la2)
head(as.data.frame(la2de),20)
write.table(la2de,file="la2de.tsv",quote=FALSE,sep="\t")

la3de <- run_de(ssla3,la3)
head(as.data.frame(la3de),20)
write.table(la3de,file="la3de.tsv",quote=FALSE,sep="\t")

la4de <- run_de(ssla4,la4)
head(as.data.frame(la4de),20)
write.table(la4de,file="la4de.tsv",quote=FALSE,sep="\t")

la5de <- run_de(ssla5,la5)
head(as.data.frame(la5de),20)
write.table(la5de,file="la5de.tsv",quote=FALSE,sep="\t")

```

## Session Information

```{r,sessioninfo}

sessionInfo()

```





